js_modules/dagster-ui/packages/ui-core/src/workspace/repoAddressAsString.ts

```diff
@@ -1,17 +1,22 @@
-import memoize from 'lodash/memoize';
-
 import {buildRepoPathForHuman, buildRepoPathForURL} from './buildRepoAddress';
 import {RepoAddress} from './types';
 
-export const repoAddressAsHumanString = memoize((repoAddress: RepoAddress) => {
+// Note: These are not memoized because the result is a primitive value and
+// there are scenarios where the repoAddress argument is a temporary object,
+// so WeakMapping the input to the output can cause the WeakMap to become
+// huge before it's garbage collected. We could use memoize() with a cache-key
+// function, but the cache-key generator would essentially BE this method,
+// which would eat up any perf gains yielded by memoizing it.
+//
+export const repoAddressAsHumanString = (repoAddress: RepoAddress) => {
```

Comment:
I'm not sure but we might be relying on strict equality somewhere and this could break that if we are.

Reply:
Can you double check that we're not doing that anywhere?

Reply:
Ahh yeah -- these all return strings thankfully, since it's a base primitive type it's safe! 

eg: 

```
const fn = () => "A"
fn() === fn() // true
```